tools/builder/debian/Dockerfile

```diff
@@ -0,0 +1,155 @@
+# Stage 1: Create the base image with all necessary components
+FROM debian:9 as vagrant-base
```

Comment:
`debian:9` is not deterministic, we should pin the hash of the image

Reply:
The image we build from this is what matters, we'd pin the hash of the resulting image in our CI and tag it, then build against that, and point our tooling to the tagged image.